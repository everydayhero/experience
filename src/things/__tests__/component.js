/* global describe it expect */
import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import comp from '../component.js'

describe('comp', () => {
  it('should allow removing props from output components to avoid warnings', () => {
    const StyledComp = comp(({ props }) => ({
      color: props.color || 'red'
    }))('div', { removeProps: ['color'] })
    const component = renderer.create(
      <StyledComp color='green' />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should remove props even when they come from a native component', () => {
    const StyledComp = comp(({ props }) => ({
      color: props.color || 'red',
      backgroundColor: 'blue'
    }))('div')
    const StyledCompInehrit = comp(({ props }) => ({
      backgroundColor: props.black || 'black'
    }))(StyledComp, { removeProps: ['black'] })
    const componentObj = shallow(
      <StyledCompInehrit black='white' />
    )
    const component = renderer.create(
      <StyledCompInehrit black='white' />
    )
    const tree = component.toJSON()
    expect(componentObj.props()).toMatchSnapshot()
    expect(tree).toMatchSnapshot()
  })
})
