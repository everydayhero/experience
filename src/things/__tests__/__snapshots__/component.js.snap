exports[`comp should allow removing props from output components to avoid warnings 1`] = `
<div
  className="cxsync-1950064219" />
`;

exports[`comp should remove props even when they come from a native component 1`] = `
Object {
  "className": "cxsync-3634499107",
  "styles": Object {
    "backgroundColor": "white",
  },
}
`;

exports[`comp should remove props even when they come from a native component 2`] = `
<div
  className="cxsync-146553031" />
`;
